integration/PhpSpec/Console/Prompter/DialogTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace integration\PhpSpec\Console\Prompter;
+
+use PhpSpec\Console\Prompter\Dialog;
+
+/**
+ * @requires function \Symfony\Component\Console\Helper\DialogHelper::askConfirmation
+ */
+class DialogTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var \Symfony\Component\Console\Helper\DialogHelper
+     */
+    private $dialogHelper;
+
+    /**
+     * @var \PhpSpec\Console\Prompter
+     */
+    private $prompter;
+
+    function setUp()
+    {
+        $this->output = $this->getMock('Symfony\Component\Console\Output\OutputInterface');
+        $this->dialogHelper = $this->getMockBuilder('Symfony\Component\Console\Helper\DialogHelper')
+                                   ->disableOriginalConstructor()->getMock();
+
+        $this->prompter = new Dialog($this->output, $this->dialogHelper);
+    }
+
+    /**
+     * @test
+     */
+    function it_is_a_prompter()
+    {
+        $this->assertInstanceOf('PhpSpec\Console\Prompter', $this->prompter);
+    }
+
+    /**
+     * @test
+     */
+    function it_can_ask_a_question_and_return_the_result()
+    {
+        $this->dialogHelper->expects($this->once())
+                           ->method('askConfirmation')
+                           ->with($this->equalTo($this->output), 'Are you sure?', true)
```

Comment:
this should use ``$this->identicalTo()`` instead, to do strict comparison for the object